<?xml version="1.0"?>

<launch>

<arg name="domain_path"	 default="$(find erl_assignment_2)/pddl/detectibot_pddl/db_domain.pddl" />
<arg name="problem_path" default="$(find erl_assignment_2)/pddl/detectibot_pddl/db_problem.pddl" />

<arg name="use_simulated_actions" default="false" />
<arg name="use_real_actions" default="true" />

<!-- knowledge base -->
<node name="rosplan_knowledge_base" pkg="rosplan_knowledge_base" type="knowledgeBase" respawn="false" output="screen">
	<param name="domain_path" value="$(arg domain_path)" />
	<param name="problem_path" value="$(arg problem_path)" />
	<param name="use_unknowns" value="false" />
</node>

<!-- problem generation -->
<include file="$(find rosplan_planning_system)/launch/includes/problem_interface.launch">
	<arg name="knowledge_base"   value="rosplan_knowledge_base" />
	<arg name="domain_path"      value="$(arg domain_path)" />
	<arg name="problem_path"     value="$(find erl_assignment_2)/data/problem_interface/problem.pddl" />
	<arg name="problem_topic"    value="problem_instance" />
</include>

<!-- planner interface -->
<include file="$(find rosplan_planning_system)/launch/includes/planner_interface.launch">
	<arg name="use_problem_topic"    value="true" />
	<arg name="problem_topic"        value="/rosplan_problem_interface/problem_instance" />
	<arg name="planner_topic"        value="planner_output" />
	<arg name="domain_path"          value="$(arg domain_path)" />
	<arg name="problem_path"         value="$(find erl_assignment_2)/data/problem_interface/problem.pddl" />
	<arg name="data_path"            value="$(find erl_assignment_2)/data/" />
	<arg name="planner_command"      value="timeout 10 $(find rosplan_planning_system)/common/bin/popf DOMAIN PROBLEM" />
</include>

<!-- plan parsing -->
<node name="rosplan_parsing_interface" pkg="rosplan_planning_system" type="pddl_simple_plan_parser" respawn="false" output="screen">
	<param name="knowledge_base" value="rosplan_knowledge_base" />
	<param name="planner_topic"  value="/rosplan_planner_interface/planner_output" />
	<param name="plan_topic"     value="complete_plan" />
</node>

<!-- plan dispatching -->
<node name="rosplan_plan_dispatcher" pkg="rosplan_planning_system" type="pddl_simple_plan_dispatcher" respawn="false" output="screen">
	<param name="knowledge_base"        value="rosplan_knowledge_base" />
	<param name="plan_topic"            value="/rosplan_parsing_interface/complete_plan" />
	<param name="action_dispatch_topic" value="action_dispatch" />
	<param name="action_feedback_topic" value="action_feedback" />
</node>

<!-- detecibot nodes -->
<node pkg="erl_assignment_2" type="cluedo_kb.py" name="cluedo_kb" output="screen"/>
<!-- <node pkg="erl2" type="simulation" name="simulation" output="screen"/> -->
<node pkg="erl2" type="my_simulation" name="my_simulation" output="screen"/>
<!-- <node pkg="erl_assignment_2" type="" name="" output="screen"/> -->

<!-- actions -->
<include file="$(find erl_assignment_2)/launch/run_simulated_actions.launch" if="$(arg use_simulated_actions)" />
<include file="$(find erl_assignment_2)/launch/run_detectibot_actions.launch" if="$(arg use_real_actions)" />

</launch>
